src/termio/shell_integration.zig

```diff
@@ -652,6 +655,19 @@ test "xdg: existing XDG_DATA_DIRS" {
     try testing.expectEqualStrings("./shell-integration:/opt/share", env.get("XDG_DATA_DIRS").?);
 }
 
+/// Setup the nushell shell integration. This works by setting
+/// XDG_DATA_DIRS so that it can be picked automatically by
+/// nushell on startup.
+/// Only implements `ssh-*` shell features. Rest are not supported.
+fn setupNu(alloc_arena: Allocator, resource_dir: []const u8, env: *EnvMap, features: config.ShellIntegrationFeatures) !void {
+    // This makes sure that `Nu` loads our integration file
+    // and wraps the `ssh` function only if the `ssh` features
+    // are enabled.
+    // Otherwise, it does not do anything.
+    if (features.@"ssh-env" or features.@"ssh-terminfo") {
```

Comment:
I'm thinking whether this is a worthwhile optimization. I'm inclined to always set up the XDG shell environment here (unconditionally) and then evaluate the shell features more precisely in the script. What do you think?

(That would eliminate `setupNu`; we could just call `setupXdgDataDirs` in the `nu` case in `setupShell`.)